In [14]:
# load vine dataset from scikit-learn

from sklearn.datasets import load_wine
import pandas as pd
# import train test split
from sklearn.model_selection import train_test_split
import torch

data = load_wine()
X = data.data

print(data.feature_names)

df = pd.DataFrame(X, columns=data.feature_names)
df.head()

y = df['alcohol']
df = df.drop("alcohol", axis=1)
df.head()

['alcohol', 'malic_acid', 'ash', 'alcalinity_of_ash', 'magnesium', 'total_phenols', 'flavanoids', 'nonflavanoid_phenols', 'proanthocyanins', 'color_intensity', 'hue', 'od280/od315_of_diluted_wines', 'proline']


malic_acid   ash  alcalinity_of_ash  magnesium  total_phenols  flavanoids  \
0        1.71  2.43               15.6      127.0           2.80        3.06   
1        1.78  2.14               11.2      100.0           2.65        2.76   
2        2.36  2.67               18.6      101.0           2.80        3.24   
3        1.95  2.50               16.8      113.0           3.85        3.49   
4        2.59  2.87               21.0      118.0           2.80        2.69   

   nonflavanoid_phenols  proanthocyanins  color_intensity   hue  \
0                  0.28             2.29             5.64  1.04   
1                  0.26             1.28             4.38  1.05   
2                  0.30             2.81             5.68  1.03   
3                  0.24             2.18             7.80  0.86   
4                  0.39             1.82             4.32  1.04   

   od280/od315_of_diluted_wines  proline  
0                          3.92   1065.0  
1                          3.40   1050.0  
2                          3.17   1185.0  
3                          3.45   1480.0  
4                          2.93    735.0

In [18]:
x_train, x_test, y_train, y_test = train_test_split(df, y, test_size=0.2, random_state=42)

In [19]:
x_train = torch.tensor(x_train.values)
x_test = torch.tensor(x_test.values)
y_train = torch.tensor(y_train.values)
y_test= torch.tensor(y_test.values)

In [20]:
x_train.shape

torch.Size([142, 12])

In [21]:
y_train.shape

torch.Size([142])

# Überlegungen zur Netzarchitektur

* Wir haben 12 Attribute / Spalten / Features
* Wir haben 142 Traingsbeispiele

Angenommen, ein HiddenLayer mit 100 Neuronen
* w1: von 12 auf 100
* w2: von 100 auf 1



In [22]:
w1 = torch.rand((12, 100), requires_grad=True)